src/pypromice/process/L1toL2.py

```diff
@@ -114,17 +112,62 @@ def toL2(
             ds['rh_i_wrt_ice_or_water'] = adjustHumidity(ds['rh_i'], ds['t_i'],
                                              T_0, T_100, ews, ei0)
 
-    # Determiune cloud cover for on-ice stations
-    cc = calcCloudCoverage(ds['t_u'], T_0, eps_overcast, eps_clear,        # Calculate cloud coverage
-                           ds['dlr'], ds.attrs['station_id'])
-    ds['cc'] = (('time'), cc.data)
-
     # Determine surface temperature
-    ds['t_surf'] = calcSurfaceTemperature(T_0, ds['ulr'], ds['dlr'],           # Calculate surface temperature
+    ds['t_surf'] = calcSurfaceTemperature(T_0, ds['ulr'], ds['dlr'],
                                           emissivity)
-    if not ds.attrs['bedrock']:
-        ds['t_surf'] = xr.where(ds['t_surf'] > 0, 0, ds['t_surf'])
+    is_bedrock = (str(ds.attrs['bedrock']).lower() == 'true')
```

Comment:
This is better because the bedrock attribute parsing is added to the beginning of the function.

However, I still can't see why it is necessary to handle a case where 'bedrock' is something else than a boolean since it should be set in L0toL1.

```suggestion

    # Handle cases where the bedrock attribute is incorrectly set
    if not 'bedrock' in ds.attrs:
        logger.warning('bedrock attribute is not set')
        ds.attrs['bedrock'] = False
    elif not is instance(ds.attrs['bedrock'], bool):
        logger.warning('bedrock attribute is not boolean: {ds.attrs["bedrock"]}')
        ds.attrs['bedrock'] =  str(ds.attrs['bedrock']).lower() == 'true'
    
    is_bedrock = ds.attrs['bedrock']
```

Reply:
So I have now moved the sanitation of the bedrock attribute to L0toL1.py:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/1136395d6110e74472e4d7a6d9e9666e5388784b/src/pypromice/process/L0toL1.py#L102-L109

In L1toL2.py the datasets should basically come from L0toL1.py and therefore should already have the good type for the bedrock attribute:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/1136395d6110e74472e4d7a6d9e9666e5388784b/src/pypromice/process/L1toL2.py#L118

In L2toL3.py the dataset can either come from L1toL2.py or from an intermediate netcdf file. So the bedrock attribute can be either boolean or string:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/1136395d6110e74472e4d7a6d9e9666e5388784b/src/pypromice/process/L2toL3.py#L44

Is it ready to merge now?